src/main/java/mate/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package mate.service;
+
+import java.util.Optional;
+import mate.lib.Injector;
+import mate.lib.Service;
+import mate.lib.exception.AuthenticationException;
+import mate.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final Injector injector = Injector.getInstance("mate");
+    private final DriverService driverService = (DriverService) injector
+            .getInstance(DriverService.class);
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverService.findByLogin(login);
+        if (driver.get().getPassword().equals(password)) {
```

Comment:
If driver was not found you will get NPE instead of AuthenticationException, left modify this check

Reply:
done